tests/test_datamodel_service.py

```diff
@@ -472,3 +473,10 @@ def test_command_creation_inside_singleton(new_mesh_session):
     meshing = new_mesh_session
     read_mesh = meshing.meshing.File.ReadMesh.create_instance()
     assert read_mesh.FileName is not None
+
+
+def test_read_ony_set_state(new_mesh_session):
+    meshing = new_mesh_session
+    assert meshing.preferences.MeshingWorkflow.CheckpointingOption.is_read_only()
+    with pytest.raises(ReadOnlyObjectError):
+        meshing.preferences.MeshingWorkflow.CheckpointingOption = "Write into memory"
```

Comment:
@mkundu1 Is meshing.preferences.MeshingWorkflow.set_state({...}) also covered?

Reply:
yes.

Reply:
OK, is it covered by the tests?

Reply:
As discussed, we shall handle this at StateEngine level in server.